automount-actions/scripts/bin/auto-mount.sh

```diff
@@ -59,25 +59,39 @@ try_mount() {
 # cleanup acts to ensure this daemon remains tidy
 cleanup () {
     _exit_status=$?
-
     sync
-
-    # TODO: parsing ls is potentially unreliable. find is an alternative but the
-    # solution is potentially nonportable. Find a cleaner way to determine
-    # mountpoints.
-    #shellcheck disable=2012
-    _uuids="$(ls -1 "$MOUNT_DIR" | tr '\n' :)"
-    [ -z "$_uuids" ] || {
-        # Unmount everthing
-        IFS=:
-        for _uuid in $_uuids; do
-            umount "${MOUNT_DIR}/${_uuid}" || continue
-            rmdir "${MOUNT_DIR}/${_uuid}"
+    
+    # Use find instead of ls for better reliability and portability
+    # This handles filenames with spaces and special characters properly
+    if [ -d "$MOUNT_DIR" ]; then
+        # Find all directories in MOUNT_DIR (these are our UUID mount points)
+        # Use -print0 and read -d '' to handle filenames with spaces/newlines
+        find "$MOUNT_DIR" -mindepth 1 -maxdepth 1 -type d -print0 | \
```

Comment:
Almost none of this is [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/) ;)